### ISSUE TITLE
FNX-5100 ⁃ [Bug] Unable to drive Fenix using Raptor

### ISSUE THREAD

**Issue by davehunt** [mozilla-mobile/fenix#1561]
## Steps to reproduce
1. Download and install Fenix with Raptor support onto device.
1. Run `mach raptor-test --test raptor-tp6m-1 --app fenix --binary org.mozilla.fenix --activity browser.BrowserPerformanceTestActivity` from mozilla-central.

### Expected behavior
Raptor page load tests should run. Fenix should launch, and after 30 seconds the first page load should occur. Once loaded, it should reload on repeat 15 times before moving onto the next test.

### Actual behavior
Fenix launches, but the page load never occurs. Eventually a timeout is reached.

### Device information
* Android device: Moto G5
* Fenix version: [20190408104652](https://queue.taskcluster.net/v1/task/MjZunj74Qbya0soNNz3kGQ/runs/0/artifacts/public/target.apk)

### Additional
Running logcat shows there is a JavaScript Error when attempting to update the tab: 

```
[JavaScript Error: "Unchecked lastError value: Error: Invalid tab ID: 0" {file: "moz-extension://6f61abcc-1fc1-4c0b-a9b7-4de79f8b16ba/runner.js" line: 398}]
```

This is referring to [this line](https://searchfox.org/mozilla-central/rev/69ace9da347adcc4a33c6fa3d8e074759b91068c/testing/raptor/webext/raptor/runner.js#398) in Raptor. I also tried calling `tabs.create` and saw the following error:

```
04-11 17:24:26.238 10018 10043 E GeckoConsole: [JavaScript Error: "window is null" {file: "chrome://geckoview/content/ext-tabs.js" line: 263}]
04-11 17:24:26.238 10018 10043 E GeckoConsole: create@chrome://geckoview/content/ext-tabs.js:263:30
04-11 17:24:26.238 10018 10043 E GeckoConsole: call/result</<@resource://gre/modules/ExtensionParent.jsm:950:49
04-11 17:24:26.238 10018 10043 E GeckoConsole: withPendingBrowser@resource://gre/modules/ExtensionParent.jsm:604:26
04-11 17:24:26.238 10018 10043 E GeckoConsole: call/result<@resource://gre/modules/ExtensionParent.jsm:949:24
04-11 17:24:26.238 10018 10043 E GeckoConsole: withTiming@resource://gre/modules/ExtensionParent.jsm:916:14
04-11 17:24:26.238 10018 10043 E GeckoConsole: call@resource://gre/modules/ExtensionParent.jsm:948:25
04-11 17:24:26.240 10018 10043 E GeckoConsole: [JavaScript Error: "Error: An unexpected error occurred" {file: "undefined" line: 0}]
```

**Comment by agi90** [comment]
This is because we don't support `browser.tabs` in GeckoView, I opened https://bugzilla.mozilla.org/show_bug.cgi?id=1543755

**Comment by davehunt** [comment]
FWIW this works in Fennec, GeckoView example, and Reference Browser

**Comment by rwood-moz** [comment]
Christian and I looked into this and figured out the issue. Raptor is launching the android app (mozdevice ADBDevice) using default intent of 'android.intent.action.Main' but for Fenix we need to specify the intent of 'android.intent.action.VIEW' instead. We tried that locally and with that change Raptor page-load runs fine on Fenix using the one tab / tabs.update. I'll file a Raptor bug and make a patch. This Fenix issue can be closed.

[0] https://searchfox.org/mozilla-central/rev/1b2636e8517aa48422ed516affe4d28cb7fa220a/testing/mozbase/mozdevice/mozdevice/adb.py#3095
